vowpalwabbit/core/src/reductions/cb/cb_explore_adf_large_action_space.cc

```diff
@@ -235,6 +257,54 @@ void cb_explore_adf_large_action_space::generate_B(const multi_ex& examples)
   }
 }
 
+bool cb_explore_adf_large_action_space::generate_AAtop(const multi_ex& examples)
+{
+  _triplets.clear();
+  AAtop.resize(examples.size(), examples.size());
+  _aatop_action_ft_vectors.clear();
+  _aatop_action_indexes.clear();
+  _aatop_action_indexes.resize(examples.size());
+  _aatop_action_ft_vectors.resize(examples.size());
+
+  for (size_t i = 0; i < examples.size(); ++i)
+  {
+    _aatop_action_ft_vectors[i].clear();
+    _aatop_action_indexes[i].clear();
+    _aatop_action_ft_vectors[i].resize(_all->weights.mask() + 1, 0.f);
+    auto& red_features =
+        examples[i]->_reduction_features.template get<VW::generated_interactions::reduction_features>();
+
+    AAtop_triplet_constructor tc(_all->weights.mask(), _aatop_action_ft_vectors[i], _aatop_action_indexes[i]);
+    GD::foreach_feature<AAtop_triplet_constructor, uint64_t, triplet_construction, dense_parameters>(
+        _all->weights.dense_weights, _all->ignore_some_linear, _all->ignore_linear,
+        (red_features.generated_interactions ? *red_features.generated_interactions : *examples[i]->interactions),
+        (red_features.generated_extent_interactions ? *red_features.generated_extent_interactions
+                                                    : *examples[i]->extent_interactions),
+        _all->permutations, *examples[i], tc, _all->_generate_interactions_object_cache);
+  }
+
+  for (size_t i = 0; i < examples.size(); ++i)
+  {
+    for (size_t j = 0; j < examples.size(); ++j)
```

Comment:
this part is still under development and may be removed so just putting everything on one branch so that it can be tested at the same time without recompiling

Reply:
made the change though anyway, thanks